Automattic/Gravatar-SDK-iOS
# QE: Avatar Rating

Closes #546 #548 

### Description

Adds support for viewing and changing the rating of an avatar in the Quick Editor.

### Testing Steps

1. Run the Demo app
2. Choose SwiftUI Demos --> Profile editor with oauth --> Tap `Open Profile Editor with OAuth flow`
3. Tap the action menu `...`
   - OBSERVE:
      - [x] The `Rating` menu appears in the menu
      - [x] The `Rating` menu shows the current rating for the image: `Rating: PG`
      - [x] The `Rating` menu has a icon of a half-filled star ⭐ 
4. Tap the `Rating` menu to show the sub-menu
   - OBSERVE:
      - [x] All rating options are visible, with descriptions (`G (General)`)
      - [x] The selected rating has a checkmark (✔️)
5. Choose a different rating
   - OBSERVE:
      - [x] The rating for the image changes on the device
      - [x] The rating for the image changes on web

Try:
- different layouts
- dark/light
- iPad
